### PR TITLE
SvelteKit & Invisibility Support

### PR DESCRIPTION
**Pull request by edde746**

### CODE REVIEW

File: src/HCaptcha.svelte [hunk 1]
```diff
@@ -41,59 +47,69 @@
...
   }
 </script>
 
 <svelte:head>
-  <script src={scriptSrc} async defer>
-  </script>
+  {#if mounted && !window?.hcaptcha}
+    <script>
+      alert(1);
+    </script>
```
Comment (mrmikardo) [inline src/HCaptcha.svelte]: It's not clear to me why this is needed - is this an artefact of devving?

File: src/HCaptcha.svelte [hunk 2]
```diff
@@ -41,59 +47,69 @@
 
   // construct the script tag for hCaptcha remote resources
   const query = new URLSearchParams({
-    recaptchacompat: reCaptchaCompat || "",
-    onload: "hcaptchaOnLoad",
+    recaptchacompat: reCaptchaCompat ? 'on' : 'off',
+    onload: 'hcaptchaOnLoad',
+    render: 'explicit',
```
Comment (mrmikardo) [inline src/HCaptcha.svelte:53]: Thanks, this is effectively a minor bugfix.

**Comment by mrmikardo** [comment]
@edde746 thanks for opening this - looks like a useful enhancement - I will review + test and aim to merge ASAP! :)

File: src/HCaptcha.svelte [hunk 3]
```diff
@@ -12,25 +12,31 @@
...
-    LIGHT = "light",
+    DARK = 'dark',
+    LIGHT = 'light',
   }
 </script>
 
 <script lang="ts">
-  import { onDestroy, createEventDispatcher, onMount } from "svelte";
+  import { onDestroy, createEventDispatcher, onMount } from 'svelte';
+  import { browser } from '$app/env';
```
Comment (mrmikardo) [inline src/HCaptcha.svelte]: This is tricky, as it assumes that the `HCaptcha` component lives inside of a SvelteKit app.

My suggestion for this is a reimplementation of the SvelteKit `browser` var;

```
  // @ts-ignore
  const browser = !import.meta.env.SSR || true;  // assume browser context by default
```

This breaks the assumption that `HCaptcha` is always inside a SvelteKit app.

Comment (mrmikardo) [reply src/HCaptcha.svelte]: How does that look to you?

File: src/HCaptcha.svelte [hunk 4]
```diff
@@ -12,25 +12,31 @@
...
 
   const dispatch = createEventDispatcher();
 
-  export let sitekey = window.sitekey || undefined;
-  export let apihost;
-  export let hl;
-  export let reCaptchaCompat;
-  export let theme;
+  export let sitekey: string = window?.sitekey || undefined;
```
Comment (mrmikardo) [inline src/HCaptcha.svelte]: When I tested this inside a SvelteKit app, I ran into problems with `window is not defined` (even with the nullish coalescing operator you've added).

My suggestion here is;
1. Leave `sitekey` `undefined` i.e. `export let sitekey: string;`
2. Define it in the `onMount()` callback, dependent on `browser` i.e. 

```
onMount(() => {
  if (browser && !sitekey) sitekey = window.sitekey;
  ...
```

That seems to solve the problem for me. What do you reckon?

File: src/HCaptcha.svelte [hunk 5]
```diff
@@ -12,25 +12,31 @@
...
+  export let apihost: string = 'https://js.hcaptcha.com/1/api.js';
+  export let hl: string = '';
+  export let reCaptchaCompat: boolean = false;
+  export let theme: CaptchaTheme = CaptchaTheme.LIGHT;
+  export let size: 'normal' | 'compact' | 'invisible' = 'normal';
```
Comment (mrmikardo) [inline src/HCaptcha.svelte:32]: Setting these defaults is a nice change!

**Review by mrmikardo — CHANGES_REQUESTED** [review]
Nice work here! Thanks for getting the ball rolling on this.

I've made a few comments regarding the implementation that I think will make this robust enough to merge. LMK what you think :)

File: src/HCaptcha.svelte (hunk 1 above)
Comment (edde746) [reply src/HCaptcha.svelte]: Yep, as you can see I force pushed over it.

File: src/HCaptcha.svelte (hunk 4 above)
Comment (edde746) [reply src/HCaptcha.svelte]: Yep, works great 👍

File: src/HCaptcha.svelte (hunk 3 above)
Comment (edde746) [reply src/HCaptcha.svelte]: Looks good, implemented in latest commit.

**Comment by edde746** [comment]
`!import.meta.env.SSR || true` will always be `true` if I'm not wrong, so I have instead set `browser` to `import.meta.env.SSR === undefined ? true : !import.meta.env.SSR`.

**Review by mrmikardo — APPROVED** [review]
This looks great!

Please squash this down into a single informative commit, force-push it, and I'll merge :)

**Comment by edde746** [comment]
> This looks great!
> 
> Please squash this down into a single informative commit, force-push it, and I'll merge :)

Done!

**Pull request merged**